File: frontend/src/data/modules.ts
import {
  LayoutDashboard,
  Truck,
  FileText,
  Tractor,
  Package,
  Wallet,
  ClipboardList,
  CloudSun,
  FlaskConical,
  Sprout,
  Users,
} from 'lucide-react'
import type { ModuleDefinition } from '../types/module'

export const MODULE_DEFINITIONS: ModuleDefinition[] = [
  {
    key: 'dashboard',
    path: '/dashboard',
    name: 'Visão Geral',
    description: 'Painel executivo com indicadores em tempo real da operação da fazenda.',
    category: 'estrategico',
    icon: LayoutDashboard,
    tags: ['monitoramento', 'kpi'],
  },
  {
    key: 'carregamento',
    path: '/carregamento',
    name: 'Carregamento de Caminhão',
    description: 'Controle de filas, pesagem e rastreio das expedições.',
    category: 'operacional',
    icon: Truck,
    tags: ['logística'],
  },
  {
    key: 'nota-fiscal',
    path: '/nota-fiscal',
    name: 'Notas Fiscais',
    description: 'Gestão de NF-e emitidas e recebidas com validação automática.',
    category: 'financeiro',
    icon: FileText,
  },
  {
    key: 'maquinas',
    path: '/maquinas',
    name: 'Máquinas e Equipamentos',
    description: 'Telemetria, manutenção e disponibilidade dos equipamentos.',
    category: 'operacional',
    icon: Tractor,
    tags: ['telemetria'],
  },
  {
    key: 'insumos',
    path: '/insumos',
    name: 'Controle de Insumos',
    description: 'Entrada, saída e rastreabilidade do estoque de insumos.',
    category: 'operacional',
    icon: Package,
  },
  {
    key: 'financeiro',
    path: '/financeiro',
    name: 'Financeiro',
    description: 'Fluxo de caixa, contas a pagar e receitas por cultura.',
    category: 'financeiro',
    icon: Wallet,
  },
  {
    key: 'atividades',
    path: '/atividades',
    name: 'Atividades',
    description: 'Planejamento diário, controle de talhões e checklist de execução.',
    category: 'operacional',
    icon: ClipboardList,
  },
  {
    key: 'meteorologia',
    path: '/meteorologia',
    name: 'Meteorologia',
    description: 'Monitoramento climático hiperlocal com alertas proativos.',
    category: 'estrategico',
    icon: CloudSun,
  },
  {
    key: 'solo',
    path: '/solo',
    name: 'Análise de Solo',
    description: 'Mapa de fertilidade, condutividade e recomendações por talhão.',
    category: 'estrategico',
    icon: FlaskConical,
  },
  {
    key: 'safra',
    path: '/safra',
    name: 'Gestão de Safras',
    description: 'Planejamento das safras, custos e produtividade comparada.',
    category: 'estrategico',
    icon: Sprout,
  },
  {
    key: 'usuarios',
    path: '/usuarios',
    name: 'Usuários',
    description: 'Administração dos usuários internos e seus perfis de acesso.',
    category: 'suporte',
    icon: Users,
  },
]

export const DEFAULT_ENABLED_MODULES = MODULE_DEFINITIONS.map((module) => module.key)

File: backend/app/core/modules.py
from dataclasses import asdict, dataclass
from typing import Literal


ModuleCategory = Literal['estrategico', 'operacional', 'financeiro', 'suporte']


@dataclass(frozen=True, slots=True)
class ModuleConfig:
    key: str
    name: str
    path: str
    category: ModuleCategory
    description: str
    tags: tuple[str, ...] = ()


MODULE_DEFINITIONS: tuple[ModuleConfig, ...] = (
    ModuleConfig(
        key='dashboard',
        name='Visão Geral',
        path='/dashboard',
        category='estrategico',
        description='Indicadores estratégicos e visão macro da operação agrícola.',
        tags=('kpi', 'monitoramento'),
    ),
    ModuleConfig(
        key='carregamento',
        name='Carregamento de Caminhão',
        path='/carregamento',
        category='operacional',
        description='Gestão de expedições, filas e rastreamento logístico.',
        tags=('logistica',),
    ),
    ModuleConfig(
        key='nota-fiscal',
        name='Notas Fiscais',
        path='/nota-fiscal',
        category='financeiro',
        description='Emissão, recepção e conferência automática de NF-e.',
    ),
    ModuleConfig(
        key='maquinas',
        name='Máquinas e Equipamentos',
        path='/maquinas',
        category='operacional',
        description='Telemetria, manutenções e disponibilidade de ativos.',
        tags=('telemetria',),
    ),
    ModuleConfig(
        key='insumos',
        name='Controle de Insumos',
        path='/insumos',
        category='operacional',
        description='Rastreabilidade de estoque e movimentos de insumos.',
    ),
    ModuleConfig(
        key='financeiro',
        name='Financeiro',
        path='/financeiro',
        category='financeiro',
        description='Fluxos financeiros, margem operacional e centros de custo.',
    ),
    ModuleConfig(
        key='atividades',
        name='Atividades',
        path='/atividades',
        category='operacional',
        description='Planejamento diário, checklists e status dos talhões.',
    ),
    ModuleConfig(
        key='meteorologia',
        name='Meteorologia',
        path='/meteorologia',
        category='estrategico',
        description='Monitoramento climático hiperlocal com alertas preventivos.',
    ),
    ModuleConfig(
        key='solo',
        name='Análise de Solo',
        path='/solo',
        category='estrategico',
        description='Mapas de fertilidade, amostragens e recomendações.',
    ),
    ModuleConfig(
        key='safra',
        name='Gestão de Safras',
        path='/safra',
        category='estrategico',
        description='Planejamento e execução completa do ciclo de safra.',
    ),
    ModuleConfig(
        key='usuarios',
        name='Usuários',
        path='/usuarios',
        category='suporte',
        description='Controle granular de perfis e permissões.',
    ),
)

MODULE_KEYS: set[str] = {module.key for module in MODULE_DEFINITIONS}


def as_json(module: ModuleConfig) -> dict[str, str | tuple[str, ...]]:
    return asdict(module)



